packages/sidenav/src/sidenav.ts

```diff
@@ -36,6 +39,88 @@ export class SideNav extends LitElement {
         this.value = event.detail.value;
     }
 
+    public constructor() {
+        super();
+        this.addEventListener('focusin', this.handleFocusin);
+        this.addEventListener('focusout', this.stopListeningToKeyboard);
+    }
+
+    private handleFocusin = (event: FocusEvent): void => {
+        this.startListeningToKeyboard();
+        if (event.target !== this) {
+            return;
+        }
+        const selected = this.querySelector('[selected]') as SideNavItem;
+        if (selected && !this.isDisabledChild(selected)) {
+            selected.focus();
+            return;
+        }
+        const items = [...this.querySelectorAll('sp-sidenav-item')];
```

Comment:
I do this a couple of times across the element, not sure if it's worth holding the PR for centralizing on a lifecycle point to cache this value...

Reply:
Can we not just use the `queryAll` decorator?

https://lit-element.polymer-project.org/api/modules/_lib_decorators_.html#queryall

Reply:
That's what I mean. I didn't realize there were so many places that used this data while deving. I'll get into adding this!

Reply:
Actually, that's not what I meant being the `queryAll` decorator only works for the `renderRoot` and not for the light DOM. Want me to get the similar work for the light DOM element management into this PR?

Reply:
Ah good point, fair point, I'd leave it in this case then.